test/es-module/test-esm-loader-hooks.mjs

```diff
@@ -0,0 +1,72 @@
+// Flags: --expose-internals
+import '../common/index.mjs';
+import ESMLoader from '../../lib/internal/modules/esm/loader.js';
```

Comment:
I don't know if you can `import` an internal module or if you would need to `createRequire` first. You should not try to resolve it as a relative URL though, otherwise it's expected it won't know what `primordials` is. Try one of the following:
```suggestion
import ESMLoader from 'internal/modules/esm/loader';
```
```suggestion
import { createRequire } from 'module';

const require = createRequire(import.meta.url);
const ESMLoader = require('internal/modules/esm/loader');
```

Reply:
If we can't `import from 'internal/'` when `--expose-internals` is set, it seems like something worth fixing on its own.

Reply:
huzzah! that trick worked @aduh95 🙇